src/builder.rs

```diff
@@ -256,12 +298,12 @@ impl ActionInfo {
     }
 }
 
-/// A builder that constructs a [`Bundle`] from a set of notes to be spent, and recipients
+/// A builder that constructs a [`Bundle`] from a set of notes to be spent, and outputs
 /// to receive funds.
 #[derive(Debug)]
 pub struct Builder {
     spends: Vec<SpendInfo>,
-    recipients: Vec<RecipientInfo>,
+    outputs: Vec<OutputInfo>,
     flags: Flags,
```

Comment:
To some extent, the flags are related to the `BundleType`. In particular, if `BundleType::Coinbase` is used, `flags.spends_enabled()` must be `true` (this check is missing from `Builder::build`; the `num_real_spends == 0` check is insufficient). So I'm not sure if this means we should be constructing to builder with `BundleType`, or ensuring we remember to add all necessary checks in `Builder::build`.

Reply:
The bundle type sort of implies the flags; what if `BundleType::Transactional` actually carried a flags payload, with the coinbase bundle type hardcoding the flags to `(false, true)`?

Reply:
And then we check the flags in `Build` rather than in the `add_*` methods?